### DISCUSSION TITLE
Per Post Sidebar Selection

### DISCUSSION THREAD

**Discussion by danemorgan** [holger1411/understrap#1331] [Ideas]
I'm wondering if anyone thinks a per post sidebar selection system such as the one Genesis uses would add value to Understrap?

**Comment by 0dp** [comment]
I have no reference point 😬
Years ago since i had my feet wet with Genesis

**Reply by danemorgan** [reply]
Basically, they have Full width, right-sidebar, left-sidebar, and dual sidebar layouts. There is a global default with a metabox for selecting them on each post. Additional layouts can be registered via hoks.

My thinking here is that with block editing picking up steam and our incorporating the wide and full alignments, providing end-users with a simple way to select an appropriate layout for a given page design could be helpful.

**Reply by bacoords** [reply]
So basically extending the 'page templates' to apply to posts, at least in the abstract? I know of this 'layout' genesis field of which you speak.

One thing I don't like about genesis + gutenberg is that some of these legacy options aren't great in the new editor experience. Because they added so many of these options in the UI via metaboxes, they've now created their own sidebar menu in the new editor, separate from the regular post settings sidebar. It's not very discoverable and it's just ONE MORE DRAWER in gutenberg to find things tucked away behind an icon somewhere.

This is more of a philosophical issue, but any UI-based setting should be carefully considered. Understrap is foremost a developer-centric theme, (at least in my view) and so many of these options should be left to developers to implement.
